### PR TITLE
Implemented some more functions, memory leak fix in paint_park_entrance_entry

### PR DESCRIPTION
**Pull request by Gericom**

### CODE REVIEW

**Comment by IntelOrca** [comment]
Could you create a dedicated rct_sprite for balloon and duck please, like how litter, peep and vehicle have been done.

**Comment by Gericom** [comment]
Okay. I will do that tomorrow.

File: src/world/sprite.c [hunk 1]
```diff
@@ -39,7 +52,45 @@ void create_balloon(int x, int y, int z, int colour)
...
+		sprite->duck.var_15 = 9;
+		int ebx = scenario_rand();
+		int ebp = ebx;
+		int edi = ebx;
+		ebp >>= 8;
+		edi >>= 16;
+		edi &= 0x3F;
+		ebp &= 0x1E;
+		targetX += ebp;
+		targetY += ebp;
```
Comment (duncanspumpkin) [inline src/world/sprite.c]: You can refactor line 64-71 to the following. Scenario rand is cheap but CS always likes to use up all of the bits before calling it again:

```
targetX += scenario_rand() & 0x3F;
targetY += scenario_rand() & 0x1E;
```

File: src/world/sprite.c [hunk 2]
```diff
@@ -39,7 +52,45 @@ void create_balloon(int x, int y, int z, int colour)
...
+			targetY = edi;
+			break;
+		case 2:
+			targetX = edi;
+			break;
+		case 3:
+			targetY = 8191 - edi;
+			break;
+		}
+		sprite->duck.var_1E = (ebx << 3) & 0xFF;
```
Comment (duncanspumpkin) [inline src/world/sprite.c]: Small bug here. ebx has been &'d with 3 before this function for the above switch statement.
Suggest changing the switch to:

```
uint8 direction = scenario_rand() & 3;
switch(direction){
...
}
var_1E = direction << 3;

**Pull request merged**